spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java

```diff
@@ -626,13 +628,13 @@ protected void handleMessageInternal(Message<?> message) throws Exception {
 				targetPartition = roundRobin() % numberOfKafkaPartitions;
 			}
 
-			if (Mode.embeddedHeaders.equals(producerProperties.getMode())) {
+			if (HeaderMode.embeddedHeaders.equals(producerProperties.getHeaderMode())) {
```

Comment:
This PR actually does exactly that (generalising the header mode). Am I missing something?